ansible_ai_connect/ai/api/views.py

```diff
@@ -1041,7 +1041,7 @@ class ChatEndpointThrottle(EndpointRateThrottle):
     permission_classes = [
         permissions.IsAuthenticated,
         IsAuthenticatedOrTokenHasScope,
-        IsRHInternalUser | IsTestUser,
+        IsRHInternalUser | IsTestUser | IsAAPUser,
```

Comment:
It is not just the ChatBot that will sit behind the Gateway.

You probably need to also update [these](https://github.com/ansible/ansible-ai-connect-service/blob/648fa372cc119232fa5ac695992987457383ee14/ansible_ai_connect/ai/api/views.py#L161-L179).

Reply:
I am not sure that the other permissions has to be changed as for the moment no feature make the used endpoints goes throw the AAP.

Reply:
After discussing this with @ldjebran we agreed my proposal is unnecessary.

VSCode continues to authenticate with and route direct to Lightspeed.